### PR TITLE
MySQL 5.6 modifications

### PR DESCRIPTION
**Pull request by meatballhat**
includes:
- setting empty password for `'root'@'localhost'`, `'root'@'127.0.0.1'`, and `'travis'@'%'`
- switching back to service name of `mysql`
- switching back to socket path(s) of `{/var/,}run/mysqld/mysqld.sock`

### CODE REVIEW

**Comment by meatballhat** [comment]
Testing on amethyst stack https://travis-ci.org/travis-infrastructure/packer-build/builds/180136670

File: cookbooks/travis_build_environment/recipes/mysql.rb [hunk 1]
```diff
@@ -20,27 +20,67 @@
...
+  Chef::Config[:file_cache_path],
+  'mysql_users_passwords.sql'
+)
+
+file mysql_users_passwords_sql do
+  content <<-EOF.gsub(/^\s+> /, '')
+    > SET old_passwords = 0;
+    > CREATE USER 'travis'@'%' IDENTIFIED BY '';
+    > SET PASSWORD FOR 'root'@'localhost' = PASSWORD('');
+    > SET PASSWORD FOR 'root'@'127.0.0.1' = PASSWORD('');
```
Comment (Lyoness) [inline cookbooks/travis_build_environment/recipes/mysql.rb]: :+1:

File: cookbooks/travis_build_environment/recipes/mysql.rb [hunk 2]
```diff
@@ -20,27 +20,67 @@
...
+
+service 'mysql' do
+  action %i(enable start)
+end
+
+bash 'setup mysql users and passwords' do
+  code "mysql -u root <#{mysql_users_passwords_sql}"
 end
 
 template "#{node['travis_build_environment']['home']}/.my.cnf" do
```
Comment (Lyoness) [inline cookbooks/travis_build_environment/recipes/mysql.rb]: :+1:

**Comment by meatballhat** [comment]
H Y P E R M E D I A https://github.com/travis-ci/packer-templates/issues/350

**Pull request merged**